codex/node.nim

```diff
@@ -61,9 +62,9 @@ type
   CodexNode* = object
     switch: Switch
     networkId: PeerId
-    blockStore: BlockStore
+    networkStore: NetworkStore
```

Comment:
So this means we are abandoning the `BlockStore` abstraction? Or moving `NetworkStore` from it?

Reply:
Network store inherits `BlockStore`, so no.

Reply:
Well yeah - technically, but here you are moving away from supporting the `BlockStore` abstraction inside Codex in favor of `NetworkStore` so any other store won't be compatible here, which in my view is "abandoning `BlockStore` abstraction"...

I mean we have already bumped into this limitation many times, so it make sense to separate `NetworkStore` out of this, but I know you were not so in favor of that, so that is why I am wondering about it.

Reply:
Thats not how interfaces and inheritance work :), we're still using the same abstraction, only a more specialized version of it. So no, we aren't moving aways from the `BlockStore` abstraction as you suggested in your original comment.